src/rules/oneLineRule.ts

```diff
@@ -27,80 +25,212 @@ export class Rule extends Lint.Rules.AbstractRule {
-        if (this.hasOption(OPTION_ELSE) && lastState !== undefined && !this.hasTrailingWhiteSpace(lastState.token)) {
-            var failure = this.createFailure(position, TypeScript.width(node.elseKeyword), Rule.ELSE_FAILURE_STRING);
-            this.addFailure(failure);
+    public visitForInStatement(node: ts.ForInStatement) {
+        this.handleIterationStatement(node);
+        super.visitForInStatement(node);
+    }
+
+    public visitWhileStatement(node: ts.WhileStatement) {
+        this.handleIterationStatement(node);
+        super.visitWhileStatement(node);
+    }
+
+    public visitBinaryExpression(node: ts.BinaryExpression) {
+        if (node.operator === ts.SyntaxKind.EqualsToken && node.right.kind === ts.SyntaxKind.ObjectLiteralExpression) {
+            var equalsToken = node.getChildAt(1);
+            var openBraceToken = node.right.getChildAt(0);
+            this.handleOpeningBrace(equalsToken, openBraceToken);
         }
+        super.visitBinaryExpression(node);
+    }
 
-        super.visitElseClause(node);
+    public visitVariableDeclaration(node: ts.VariableDeclaration) {
+        var initializer = node.initializer;
+        if (initializer != null && initializer.kind === ts.SyntaxKind.ObjectLiteralExpression) {
+            var equalsToken = node.getChildAt(1);
+            var openBraceToken = initializer.getChildAt(0);
+            this.handleOpeningBrace(equalsToken, openBraceToken);
+        }
+        super.visitVariableDeclaration(node);
     }
 
-    public visitCatchClause(node: TypeScript.CatchClauseSyntax): void {
-        var lastState = this.getLastState();
-        var position = this.getPosition() + TypeScript.leadingTriviaWidth(node);
-        if (this.hasOption(OPTION_CATCH) && lastState !== undefined && !this.hasTrailingWhiteSpace(lastState.token)) {
-            var failure = this.createFailure(position, TypeScript.width(node.catchKeyword), Rule.CATCH_FAILURE_STRING);
-            this.addFailure(failure);
+    public visitDoStatement(node: ts.DoStatement) {
+        var doKeyword = node.getChildAt(0);
+        var statement = node.statement;
+        if (statement.kind === ts.SyntaxKind.Block) {
+            var openBraceToken = statement.getChildAt(0);
+            this.handleOpeningBrace(doKeyword, openBraceToken);
         }
+        super.visitDoStatement(node);
+    }
 
-        super.visitCatchClause(node);
+    public visitModuleDeclaration(node: ts.ModuleDeclaration) {
+        var nameNode = node.name;
+        var body = node.body;
+        if (body.kind === ts.SyntaxKind.ModuleBlock) {
+            var openBraceToken = body.getChildAt(0);
+            this.handleOpeningBrace(nameNode, openBraceToken);
+        }
+        super.visitModuleDeclaration(node);
+    }
+
+    public visitEnumDeclaration(node: ts.EnumDeclaration) {
+        var nameNode = node.name;
+        var openBraceToken = node.getChildren().filter((child) => child.kind === ts.SyntaxKind.OpenBraceToken)[0];
```

Comment:
use a private utility method that filters children based on the kind and gives you back the first child. also, this is slightly buggy because it'll crash TSLint when indexing into [0] if the child is not present. we shouldn't be crashing even in incorrect code.